### PR TITLE
Create FenceCommentsOp and use it to simplify some confusing comment handling situations

### PR DESCRIPTION
**Pull request by nreid260**

### CODE REVIEW

**Comment by nreid260** [comment]
https://github.com/facebook/ktfmt/issues/380

**Comment by cgrushko** [comment]
@nreid260  thanks for the PR. Is this also relevant to GJF?

**Comment by nreid260** [comment]
This might be useful, I'll see what the upstream owners think. We can always revert this later if it lands in gjf.

File: core/src/main/java/com/facebook/ktfmt/format/FenceCommentsOp.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.facebook.ktfmt.format
```
Comment (cgrushko) [inline core/src/main/java/com/facebook/ktfmt/format/FenceCommentsOp.kt:1]: @nreid260 , this file needs a copyright header.
But I don't know what to write there 🤷 I think the easiest is to put both Google and Meta (i.e., two stacked headers)

Comment (nreid260) [reply core/src/main/java/com/facebook/ktfmt/format/FenceCommentsOp.kt:17]: Added header. I don't see this as particularly different than code I've written in other files.

**Pull request closed without merging**